test/unit/index.test.ts

```diff
@@ -26,74 +26,179 @@
+import * as ThirdPartyRequestAuth from '~/server/handlers/thirdpartyRequests/transactions/{ID}/authorizations'
+import * as ConsentsDomain from '~/domain/consents'
+
+// Mock data
+import MockConsentData from './data/mockConsent.json'
+import MockUpdateConsentReq from './data/mockUpdatedConsent.json'
+import MockGenerateChallengeReq from './data/mockGenerateChallenge.json'
+import MockThirdPartyAuthorizationReq from './data/mockThirdPartyReqAuth.json'
+import Headers from './data/headers.json'
+
+const mockLoggerPush = jest.spyOn(Logger, 'push')
+const mockLoggerError = jest.spyOn(Logger, 'error')
 
 describe('index', (): void => {
   it('should have proper layout', (): void => {
     expect(typeof index.server).toBeDefined()
     expect(typeof index.server.run).toEqual('function')
   })
+})
+
+describe('api routes', (): void => {
+  let server: Server
+
+  beforeAll(async (): Promise<Server> => {
+    mockLoggerPush.mockReturnValue(null)
+    mockLoggerError.mockReturnValue(null)
+    server = await index.server.run(Config)
+    return server
+  })
+
+  afterAll(async (done): Promise<void> => {
+    server.events.on('stop', done)
+    await server.stop()
+    jest.clearAllMocks()
+  })
+
+  beforeEach((): void => {
+    jest.clearAllMocks()
+  })
+
+  it('/health', async (): Promise<void> => {
+    interface HealthResponse {
+      status: string;
+      uptime: number;
+      startTime: string;
+      versionNumber: string;
+    }
+
+    const request = {
+      method: 'GET',
+      url: '/health'
+    }
+
+    const response = await server.inject(request)
+    expect(response.statusCode).toBe(200)
+    expect(response.result).toBeDefined()
+
+    const result = response.result as HealthResponse
+    expect(result.status).toEqual('OK')
+    expect(result.uptime).toBeGreaterThan(1.0)
+  })
+
+  it('/hello', async (): Promise<void> => {
+    interface HelloResponse {
+      hello: string;
+    }
+
+    const request = {
+      method: 'GET',
+      url: '/hello'
+    }
+
+    const response = await server.inject(request)
+    expect(response.statusCode).toBe(200)
+    expect(response.result).toBeDefined()
+
+    const result = response.result as HelloResponse
+    expect(result.hello).toEqual('world')
+  })
+
+  it('/metrics', async (): Promise<void> => {
+    const request = {
+      method: 'GET',
+      url: '/metrics'
+    }
+
+    const response = await server.inject(request)
+    expect(response.statusCode).toBe(200)
+    expect(response.result).toBeDefined()
+  })
+
+  it('POST /consents/', async (): Promise<void> => {
+    const mockPostConsent = jest.spyOn(ConsentsDomain, 'createAndStoreConsent')
+    mockPostConsent.mockResolvedValueOnce()
+
+    const request = {
+      method: 'POST',
+      url: '/consents',
+      headers: Headers,
+      payload: MockConsentData.payload
+    }
+
+    const response = await server.inject(request)
+    expect(response.statusCode).toBe(202)
+    expect(response.result).toBeDefined()
```

Comment:
We should also check that `mockPostConsent` was called once, and also verify that what it was called with matches what we expect.

Reply:
(same goes for all the others)